akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/PredefinedFromStringUnmarshallers.scala

```diff
@@ -78,11 +66,39 @@ trait PredefinedFromStringUnmarshallers {
       }
     }
 
-  val CsvString: Unmarshaller[String, immutable.Seq[String]] =
+  val CsvStringSeq: Unmarshaller[String, immutable.Seq[String]] =
     Unmarshaller.strict[String, immutable.Seq[String]] { string ⇒
       string.split(",").toList
     }
 
+  val CsvByteSeq: Unmarshaller[String, immutable.Seq[Byte]] =
+    CsvStringSeq.map(_.map(byteFromString))
+
+  val CsvShortSeq: Unmarshaller[String, immutable.Seq[Short]] =
+    CsvStringSeq.map(_.map(shortFromString))
+
+  val CsvIntSeq: Unmarshaller[String, immutable.Seq[Int]] =
+    CsvStringSeq.map(_.map(intFromString))
+
+  val CsvLongSeq: Unmarshaller[String, immutable.Seq[Long]] =
+    CsvStringSeq.map(_.map(longFromString))
```

Comment:
The problem is with this bit:
```scala
  val CsvStringSeq: Unmarshaller[String, immutable.Seq[String]] =
     Unmarshaller.strict[String, immutable.Seq[String]] { string ⇒
       string.split(",").toList
     }
```
This is not, in fact a proper CSV unmarshaller; the only CSV sequences it can decode are those that don't include commas. So, for example, a CSV-encoded sequence of Point objects might look like:

`"5,3","0,0","0,1"`

But the unmarshaller above would split it to:

```
"5
3"
"0
0"
"0
1"
```

This is obviously undesirable, so even having a so-called 'CsvString' unmarshaller that does the above is questionable, but perhaps tolerable if we expect the user to exercise due caution. Having 'CsvIntSeq' is no more problematic, as Ints certainly don't contain commas.

Having a generic CSV unmarshaller that can accept any arbitrary element unmarshaller, but that will in fact fail for a wide range of element formats, is probably a very bad idea.

So, short of adding a full-fledged CSV unmarshalling library - probably not _that_ much effort - I'm not sure it's possible to take it far beyond this.